app/src/ui/changes/no-changes.tsx

```diff
@@ -318,33 +318,43 @@ export class NoChanges extends React.Component<
       return null
     }
 
-    const itemId: MenuIDs = 'open-external-editor'
-    const menuItem = this.getMenuItemInfo(itemId)
+    return this.tryRenderMenuBackedAction(
+      'open-external-editor',
+      'external editor'
+    )
+  }
+
+  private renderOpenInShell() {
+    return this.tryRenderMenuBackedAction('open-in-shell', 'shell')
+  }
+
+  private tryRenderMenuBackedAction(menuId: MenuIDs, whatToOpen: string) {
```

Comment:
I dont love `whatToOpen` but I couldn't determine come up with a better variable name at the moment.

Reply:
`programType`?